File: Src/ButterflyNet.Parser/Scopes/ParagraphScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class ParagraphScope : ScopeBase {
		public override ScopeLevel Level { get { return ScopeLevel.Block; } }
		public override bool ClosesOnContext { get { return true; } }

		protected override void OpenAndAnalyze(ButterflyAnalyzer analyzer) {
			analyzer.OpenParagraph();
		}

		protected override void CloseAndAnalyze(ButterflyAnalyzer analyzer) {
			analyzer.CloseParagraph();
		}
	}
}

File: src/js/src/scopes/ParagraphScope.js
function ParagraphScope() {
	this.type = ScopeTypeCache.paragraph;
	this.level = ScopeLevel.block;
	this.canNestText = true;
	this.open = function(analyzer) {
		analyzer.openParagraph();
	};
	this.close = function(analyzer) {
		analyzer.closeParagraph();
	};
}

extend(Scope, ParagraphScope);